sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/conditionalExpressions.scala

```diff
@@ -262,37 +262,91 @@ case class CaseWhenCodegen(
     //     }
     //   }
     // }
+
+    val isNull = ctx.freshName("caseWhenIsNull")
+    val value = ctx.freshName("caseWhenValue")
+
     val cases = branches.map { case (condExpr, valueExpr) =>
```

Comment:
Can we follow what we did for `And` and `Or`, and just check the code length at the beginning? TBH I don't understand your change after several minutes reading.

Reply:
For `CaseWhen`, the code bloat occurs in one case class. The `CaseWhenCodegen.doGenCode` can generate deeply-nested `if-then-else` statements as above in the comment. Each element in `cases` has only a `if-then`.  Thus, it is not possible to insert code check here. Since `And` and `Or` generates deeply nested `if-then-else` by calling `doGenCode` many times, to check code size here works well.

[This line](https://github.com/apache/spark/pull/18641/files#diff-a966ee88604a834221e82916ec051d7dR286) generates the nested `if-then-else`. Thus, after this line, code size check is performed.

What do you think?

Reply:
We only codegen CASE WHEN if the case branches are less than 20, I think check code size here is good enough.

Reply:
I see. Do you want to add such a code?
```
val cases = ...
val concatCases = cases.mkString("\nelse {\n")
if (concatCases.length > 1024) {
  // current code
  var isGlobalVariable = false
  ...
}
```